src/main.ts

```diff
@@ -15,16 +15,23 @@ import * as pulumiCli from './libs/pulumi-cli';
 
 const main = async () => {
   const config = await makeConfig();
+  const usingPulumiBackend = config.cloudUrl == '' || config.cloudUrl.startsWith('https');
+
   core.debug('Configuration is loaded');
+  core.debug(`Backend is ${config.cloudUrl || "Pulumi"}`);
 
   await pulumiCli.downloadCli(config.options.pulumiVersion);
 
-  if (environmentVariables.PULUMI_ACCESS_TOKEN !== '') {
-    core.debug(`Logging into Pulumi`);
-    await pulumiCli.run('login');
-  } else if (config.cloudUrl) {
+  if (usingPulumiBackend && environmentVariables.PULUMI_ACCESS_TOKEN == '') {
+    throw new Error('PULUMI_ACCESS_TOKEN must be set to login to the Pulumi backend');
+  }
```

Comment:
This looks correct since any Url starting with `https` _should_ be a Pulumi service. However, it's _technically possible_ to login without providing PULUMI_ACCESS_TOKEN, either by passing the access token via stdin, or by providing a `credentials.json` file in `$PULUMI_HOME`. While it's probably not going to break other users, I'd prefer to not introduce an exception here.

I think it's possible for `login` to succeed if you don't